server/src/domain/system-config/system-config.service.spec.ts

```diff
@@ -269,13 +262,14 @@ describe(SystemConfigService.name, () => {
   });
 
   describe('updateConfig', () => {
-    it('should notify the microservices process', async () => {
+    it('should update the config and emit client and server events', async () => {
       configMock.load.mockResolvedValue(updates);
 
       await expect(sut.updateConfig(updatedConfig)).resolves.toEqual(updatedConfig);
 
+      expect(communicationMock.broadcast).toHaveBeenCalled();
+      expect(communicationMock.sendServerEvent).toHaveBeenCalledWith(ServerEvent.CONFIG_UPDATE);
       expect(configMock.saveAll).toHaveBeenCalledWith(updates);
-      expect(jobMock.queue).toHaveBeenCalledWith({ name: JobName.SYSTEM_CONFIG_CHANGE });
     });
```

Comment:
Can this test just connect to the channel and assert it receives the message?

Reply:
If this is a unit test I think this is the right level to mock and assert. We should have more tests on the event emitters themselves to ensure that we are confident that code works.

Reply:
It is a unit test and I think it makes sense like we have it. e2e will definitely take some more effort to check in an automated fashion, but I have at least manually verified that this is working correctly.